IdentityCore/src/workplacejoin/MSIDWorkPlaceJoinUtilBase.m

```diff
@@ -380,6 +381,13 @@ + (MSIDWPJKeyPairWithCert *)getWPJKeysWithTenantId:(__unused NSString *)tenantId
     {
         defaultKeys.keyChainVersion = MSIDWPJKeychainAccessGroupV2;
         MSID_LOG_WITH_CTX(MSIDLogLevelInfo, context, @"Returning EC private device key from default registration.");
+        // Query the session transport key only for iOS.
+        // 1P apps use transport key to decrypt ECDH JWE responses when redeeming bound regular refresh tokens
+        id keyType = privateKeyAttributes[(__bridge id)kSecAttrKeyType];
+        if (keyType && [keyType isEqual: (__bridge id)kSecAttrKeyTypeECSECPrimeRandom])
+        {
+            defaultKeys.privateTransportKeyRef = [self getSessionTransportKeyFromSecureEnclaveForTenantId:tenantId context:context];
```

Comment:
can we also try to re-purpose `findWPJRegistrationInfoWithAdditionalPrivateKeyAttributes` to return STK PrvKey Ref instead of adding to defaultKeys. This makes one function return keys+cert and don't require privateTransportKeyRef to be re-appended to defaultKeys.